### ISSUE TITLE
Postmix account won't load - no way to recover labels etc. from GUI

### ISSUE THREAD

**Issue by MaverickJones** [sparrowwallet/sparrow#801]
I'm running Sparrow 1.7.0 on NixOS.  This issue is regarding a whirlpool wallet.

I've seen this happen occasionally, I think even before 1.7.0, and usually some combination of closing the wallet, closing the app, closing both, and reopening/decrypting eventually results in all 4 whirlpool wallets showing up.  However, when it fails, it always fails on loading the Postmix account, and the log file shows the below exception.  

As I said, it's been intermittent, but currently nothing I do (including rebooting) seems to knock it through so they all load. 

Here's the exception:

`2023-01-04 09:44:09,339 ERROR [JavaFX Application Thread] c.s.s.AppController [null:-1] Error opening wallet
java.util.ConcurrentModificationException: null
	at java.base/java.util.TreeMap$PrivateEntryIterator.nextEntry(TreeMap.java:1486)
	at java.base/java.util.TreeMap$KeyIterator.next(TreeMap.java:1540)
	at com.sparrowwallet.drongo/com.sparrowwallet.drongo.wallet.WalletNode.copy(Unknown Source)
	at com.sparrowwallet.drongo/com.sparrowwallet.drongo.wallet.Wallet.copy(Unknown Source)
	at com.sparrowwallet.sparrow@1.7.0/com.sparrowwallet.sparrow.io.Storage.restorePublicKeysFromSeed(Unknown Source)
	at com.sparrowwallet.sparrow@1.7.0/com.sparrowwallet.sparrow.AppController.openWallet(Unknown Source)
	at com.sparrowwallet.sparrow@1.7.0/com.sparrowwallet.sparrow.AppController.openWallet(Unknown Source)
	at com.sparrowwallet.sparrow@1.7.0/com.sparrowwallet.sparrow.AppController.lambda$openWalletFile$41(Unknown Source)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.concurrent.EventHelper.fireEvent(EventHelper.java:219)
	at javafx.graphics/javafx.concurrent.Service.fireEvent(Service.java:856)
	at javafx.graphics/javafx.concurrent.Service.lambda$new$4(Service.java:556)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(ExpressionHelper.java:181)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.fireValueChangedEvent(ObjectPropertyBase.java:106)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.markInvalid(ObjectPropertyBase.java:113)
	at javafx.base/javafx.beans.property.ObjectPropertyBase$Listener.invalidated(ObjectPropertyBase.java:234)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper$SingleInvalidation.fireValueChangedEvent(ExpressionHelper.java:136)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.fireValueChangedEvent(ObjectPropertyBase.java:106)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.markInvalid(ObjectPropertyBase.java:113)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.set(ObjectPropertyBase.java:147)
	at javafx.graphics/javafx.concurrent.Task.setState(Task.java:698)
	at javafx.graphics/javafx.concurrent.Task$TaskCallable.lambda$call$1(Task.java:1437)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:290)
	at java.base/java.lang.Thread.run(Thread.java:833)
`

Any other details that might help, let me know.  Any assistance would be appreciated.  My primary goal here is to recover the labels from the Postmix account; getting the current wallet file to consistently load properly so I can keep remixing the whirlpool UTXOs would definitely be a bonus. 

Thanks!

**Comment by MaverickJones** [comment]
Some new info:  was trying some different things just now, and one thing I did was try to open the same wallet while it was already opened.  It prompted for the encryption password, showed the decrypting animation, and then no other effect.  I closed the wallet (but left the app open), re-opened it, redecyrypted it, and THIS TIME, it loaded all accounts except PREMIX.  The premix account had the same exception stack.

Good news is:  I now have labels and tx data for all of my accounts in CSV (excellent work thinking of that export CSV feature!!)

Still hoping the call stack sheds some light, hope this extra context helps.
